tensorrt_llm/_torch/visual_gen/models/glm_image/pipeline_glm_image.py

```
@@ -0,0 +1,1101 @@
+            token_ids.float(), scale_factor=2, mode="nearest"
+        ).to(dtype=torch.long)
+        token_ids = token_ids.view(1, -1)
+        return token_ids
+
+    @property
+    def do_classifier_free_guidance(self):
+        return self._guidance_scale > 1
+
+    def _get_glyph_embeds(
+        self,
+        prompt: Union[str, List[str]] = None,
```

Ruff (0.15.20)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)